### ISSUE TITLE
traveling_salesman_problem seems to ignore the weight parameter

### ISSUE THREAD

**Issue by dt99jay** [networkx/networkx#7720]
### Current Behavior

`traveling_salesman_problem` returns long paths when `weight` is named something other than 'weight'.

### Expected Behavior

`traveling_salesman_problem` should use the `weight` param to allow the correct edge attribute to be selected.

### Steps to Reproduce

```
import shapely
import networkx as nx
import matplotlib.pyplot as plt

# Make a 10x10 grid

vert = shapely.geometry.MultiLineString([[(x, 0), (x, 100)] for x in range(0, 110, 10)])
hori = shapely.affinity.rotate(vert, 90)
grid = shapely.unary_union([vert, hori])

params = ["weight", "distance"]
paths = []

for param in params:

    # Turn it into a graph

    graph = nx.Graph()
    graph.add_edges_from([(*line.coords, {param: line.length}) for line in grid.geoms])
    
    # Select nodes and visit them via TSP
    
    nodes = [(20., 20.), (30., 30.), (20., 80.), (80., 20.), (50., 50.), (60., 10.), (40., 40.), (50., 40.), (50, 30)]
    
    path = nx.approximation.traveling_salesman_problem(
        graph,
        weight=param,
        nodes=nodes,
        cycle=False,
        method=nx.approximation.christofides
    )
    
    paths.append(shapely.geometry.LineString(path))

print([path.length for path in paths])

```

> [240.0, 340.0]

### Environment

Python version: 3.11 & 3.10.12
NetworkX version: 3.2.1 & 3.4.2

### Additional context
```
# Plot results

fig, axes = plt.subplots(1, 2, figsize=(10, 5), sharey=True)

for ax, param, path, c in zip(axes, params, paths, ["b", "r"]):
    for line in grid.geoms:
        ax.plot(*line.xy, c="k", lw=.25)
    ax.scatter(*zip(*nodes), c="k")
    ax.plot(*path.xy, c=c)
    ax.set_title(f"Param={param}, length={path.length}")
    ax.set_aspect("equal")
```
![image](https://github.com/user-attachments/assets/ef5dd219-997a-416f-b934-75ca1e5a5516)

**Comment by dt99jay** [comment]
Thanks! Is the fix likely to be in 3.4.3?

**Comment by mjschwenne** [comment]
I would expect this to be in the next release, but I don't know when that will happen or what version number that will be.
